spring-vault-core/src/main/java/org/springframework/vault/authentication/LifecycleAwareSessionManager.java

```diff
@@ -221,10 +223,10 @@ protected boolean renewToken() {
 				return false;
 			}
 
-			throw new VaultException(VaultResponses.getError(e.getResponseBodyAsString()));
+			throw new VaultHttpException(VaultResponses.getError(e.getResponseBodyAsString()), e.getStatusCode());
```

Comment:
Not exactly sure, but that could be a place for a more specific exception like a `VaultSessionException`. Same applies for the code below. Retaining the original exception would be a good overall improvement.

Reply:
Sure.